### PR TITLE
name and type in descriptor(s) is ignored via declare_parameter(s).

### PR DESCRIPTION
**Pull request by fujitatomoya**
address https://github.com/ros2/rclpy/issues/879

Signed-off-by: Tomoya Fujita <Tomoya.Fujita@sony.com>

### CODE REVIEW

**Comment by fujitatomoya** [comment]
@ivanpauno follow up for https://github.com/ros2/rclpy/issues/879, could you take a look?

File: rclpy/rclpy/node.py [hunk 1]
```diff
@@ -349,6 +349,9 @@ def declare_parameter(
         This method, if successful, will result in any callback registered with
         :func:`add_on_set_parameters_callback` to be called.
 
+        The name and type in the given descriptor is ignored, and should be specified using
+        the name argument to this function and the default value's type instead.
```
Comment (ivanpauno) [inline rclpy/rclpy/node.py:353]: I think we need a tutorial about typed parameters without a default value and how to get things like an empty array of strings as a default value, which it's not well documented and it's a bit tricky.
But that's unrelated to this contribution.

**Review by ivanpauno — APPROVED** [review]
LGTM!

**Comment by ivanpauno** [comment]
Rpr failures are unrelated and the changes are only in the docs, so merging!

**Pull request merged**